### PR TITLE
DDFS integration

### PR DESCRIPTION
**Pull request by acrellin**
Finally!

### CODE REVIEW

File: mltsp/parallel_processing.py [hunk 1]
```diff
@@ -422,12 +453,26 @@ def featurize_in_parallel(headerfile_path, zipfile_path, features_to_use=[],
     zipfile = tarfile.open(zipfile_path)
     zipfile.extractall(tmp_dir_path)
     all_fnames = zipfile.getnames()
+    all_fnames = [f for f in all_fnames if not os.path.isdir(f)]
+    if is_test:
+        all_fnames = all_fnames[:3]
+
+    orig_fnames_dict = {}
```
Comment (stefanv) [inline mltsp/parallel_processing.py]: This code looks very much like the one above.  Perhaps useful to split out?

File: mltsp/parallel_processing.py [hunk 2]
```diff
@@ -422,12 +453,26 @@ def featurize_in_parallel(headerfile_path, zipfile_path, features_to_use=[],
...
+    tags = []
+    for i in range(len(all_fnames)):
+        short_fname = ntpath.basename(all_fnames[i])
+        tags.append(str(session_key +
+                        short_fname.replace(".", "_")))
+        orig_fnames_dict[short_fname.replace(".", "_")] = short_fname
+        if not os.path.isabs(all_fnames[i]):
+            all_fnames[i] = os.path.join(tmp_dir_path, all_fnames[i])
+    # Push all data files to DDFS
+    print(all_fnames)
```
Comment (stefanv) [inline mltsp/parallel_processing.py]: Debug statement, or meant to be here?  If so, perhaps annotate with some user-readable prefix.

File: mltsp/parallel_processing.py [hunk 3]
```diff
@@ -446,15 +488,20 @@ def featurize_in_parallel(headerfile_path, zipfile_path, features_to_use=[],
...
+        input_list=tags, params=params)
 
     fname_features_dict = {}
     for k, v in disco_results:
         fname_features_dict[k] = v
 
-    os.remove(f.name)
     print("Done generating features.")
+    disco_tools.delete_pushed_objects(session_key)
+    for key in fname_features_dict.keys():
```
Comment (stefanv) [inline mltsp/parallel_processing.py]: ``for key in fname_features:``

File: mltsp/parallel_processing.py [hunk 4]
```diff
@@ -446,15 +488,20 @@ def featurize_in_parallel(headerfile_path, zipfile_path, features_to_use=[],
...
+        fname_features_dict[orig_fnames_dict[key]] = fname_features_dict[key]
```
Comment (stefanv) [inline mltsp/parallel_processing.py]: Or perhaps:

```
for (key, val) in fname_features_dict:
    fname_features_dict[orig_fnames_dict[key]] = value
```

**Comment by stefanv** [comment]
Fantastic work, Ari!

The tests are failing because Disco is missing from the test image.  Let me get that installed and retrigger the build.

File: mltsp/parallel_processing.py (hunk 4 above)
Comment (acrellin) [inline mltsp/parallel_processing.py]: You mean `fname_features_dict.items()`? Yes, that's cleaner. Thanks.

**Comment by acrellin** [comment]
OK, so I found some holes in my unit tests - this shouldn't be passing yet... I may not have time to fix it this weekend before I leave. So close though!

**Comment by stefanv** [comment]
Disco is activated on the base image,and it looks like your earlier commit passed!

**Comment by acrellin** [comment]
Ok, I'll try to make some time this weekend to get it passing before I
leave Sunday (through the following Sunday).

**Comment by stefanv** [comment]
Thanks, Ari, that would put me in a great position to continue on the local Python worker + frontend.

**Comment by acrellin** [comment]
Green light!

**Comment by acrellin** [comment]
Maybe we should look into if there's a way to have a list of files to which changes should not trigger a new Drone build. E.g. `TODO.md`, `README.md`, etc.

**Comment by stefanv** [comment]
This puts us on the right track for the MVP.  Thanks, Ari!

**Pull request merged**